### PR TITLE
TS migration for Domain Architectures

### PR DESCRIPTION
**Pull request by gustavo-salazar**
TypeScript and visual framework migrations for the Domain architecture pages. 

I'll work on the Download All DA, but I thought it would be easier to review if we separate the TS migration beforehand.

This doesn't include the interactive Search by IDA area, might do that one after the Download part if there is time left in this sprint.

### CODE REVIEW

File: src/subPages/SimilarProteins/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+      <div className={css('similar-proteins-selector-panel')}>
+        <p>The below table lists the similar proteins from</p>
+        <Tooltip title="Switch to view similar proteins from UniProt or Reviewed databases">
+          <ToggleSwitch
+            switchCond={similarProtDb === 'uniprot'}
+            name={'proteinDB'}
+            id={'proteinDB-input'}
+            SRLabel={'View proteins from'}
+            onValue={'UniProt'}
```
Comment (matthiasblum) [inline src/subPages/SimilarProteins/index.tsx:49]:
```suggestion
            onValue={'UniProtKB'}
```

File: src/subPages/SimilarProteins/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            offValue={'Reviewed'}
```
Comment (matthiasblum) [inline src/subPages/SimilarProteins/index.tsx:50]:
```suggestion
            offValue={'Reviewed (Swiss-Prot)'}
```

File: src/subPages/SimilarProteins/index.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  return (
+    <div className={css('vf-stack', 'vf-stack--400')}>
+      <SimilarProteinsHeader
+        accession={proteinAccession || ''}
+        databases={(dataBase.payload && dataBase.payload.databases) || {}}
+      />
+
+      <div className={css('similar-proteins-selector-panel')}>
+        <p>The below table lists the similar proteins from</p>
```
Comment (matthiasblum) [inline src/subPages/SimilarProteins/index.tsx:42]:
```suggestion
        <p>The table below lists similar proteins from</p>
```

File: src/subPages/SimilarProteins/Table/index.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                  description: {
+                    main: { key: 'protein' },
+                    protein: { db: sourceDatabase, accession: acc },
+                  },
+                }}
+              >
+                {acc}
+              </Link>{' '}
+              {sourceDatabase === 'reviewed' && (
+                <Tooltip title="Reviewed by UniProt curators (Swiss-Prot)">
```
Comment (matthiasblum) [inline src/subPages/SimilarProteins/Table/index.tsx:164]:
```suggestion
                <Tooltip title="Reviewed by UniProtKB curators">
```

File: src/subPages/SimilarProteins/Table/index.tsx [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+                'vf-button--sm',
+                'generate-button',
+              )}
+            >
+              <span
+                className={css('icon', 'icon-common', 'icon-export')}
+                data-icon="&#xf233;"
+              />{' '}
+              <span className={css('file-label')}>Web View</span>
+            </Link>
```
Comment (matthiasblum) [inline src/subPages/SimilarProteins/Table/index.tsx:142]: Has a slightly different look that other buttons. Could be make exactly like other buttons, if possible.

File: src/subPages/SimilarProteins/Header/index.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  if (loading || loadingDomain) return <Loading />;
+  if (!payload || !payloadDomain) return null;
+  const representative = formatDomainsPayload(payloadDomain, payload.ida);
+  if (!representative) return null;
+  const idaObj = ida2json(payload.ida, representative, idaAccessionDB);
+  return (
+    <div>
+      <header>
+        All the proteins in this page share the domain architecture below with
+        the protein with accession <b>{accession}</b>.
```
Comment (matthiasblum) [inline src/subPages/SimilarProteins/Header/index.tsx:98]:
```suggestion
        All proteins featured on this page exhibit the same 
        domain architecture as the protein identified 
        by the accession <b>{accession}</b>.
```

File: src/components/Matches/ProtVistaMatches.js [hunk 7]
```diff
@@ -1,3 +1,4 @@
+// TODO: DELETE after checking is not been used anymore (currently ony genome3d) and that will dissapear in another PR
```
Comment (matthiasblum) [inline src/components/Matches/ProtVistaMatches.js:1]: #550 can be merged, then `IDA-TS-download` can be rebased onto `dev`.

File: src/components/Entry/DomainArchitectures/index.tsx [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+              ))}
+            </ul>
+          </div>
+        ) : null}
+      </div>
+    );
+  } else {
+    messageContent = (
+      <>
+        <h4>{payload.count} domain architectures found.</h4>
```
Comment (matthiasblum) [inline src/components/Entry/DomainArchitectures/index.tsx:202]:
```suggestion
        <h4>{payload.count} domain architectures</h4>
```

File: src/components/Entry/DomainArchitectures/index.tsx [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+            <b>Pfam</b> accessions:
+            <ul>
+              {invalidEntries.map((e) => (
+                <li key={e}>{e}</li>
+              ))}
+            </ul>
+          </div>
+        ) : null}
+      </div>
+    );
```
Comment (matthiasblum) [inline src/components/Entry/DomainArchitectures/index.tsx:198]:
```suggestion
    messageContent = (
      <div className={css('callout', 'warning')}>
        No domain architectures found. Domain architectures are calculated for InterPro entries of type Domain.
        Please ensure that the entries you are searching are of type Domain.
      </div>
    );
```

Changes:

- Use a _warning_ callout
- Update message
- Remove the bit with _Please check if the following are valid_, as it shows a list with one empty element (see screenshot)

image

File: src/components/Entry/DomainArchitectures/Options/index.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+      idaAccessionDB === 'pfam' ? 'interpro' : 'pfam',
+    );
+  };
+
+  return (
+    <nav className={css('options-panel')}>
+      <div className={css('accession-selector-panel')}>
+        <label>
+          Database:{' '}
+          <Tooltip title="Toogle between domain architectures based on Pfam and InterPro entries">
```
Comment (matthiasblum) [inline src/components/Entry/DomainArchitectures/Options/index.tsx:33]:
```suggestion
          <Tooltip title="Switch between domain architectures based on Pfam and InterPro entries">
```

**Review by matthiasblum — COMMENTED** [review]
Looks good. Made a few suggestions, but nothing major.
One thing I noticed is that when hovering domains in the Domain Architectures subpages (e.g. `/interpro/entry/InterPro/IPR000001/domain_architecture/`, the popup first appears on the top left on the viewport before being correctly placed next to the domain, e.g.:

Screenshot_2023-12-13_15-43-39

Can you see if you can reproduce this behavior, and if so, can it be fixed?

**Comment by gustavo-salazar** [comment]
OK suggestions implemented. Except for the thing about the tooltip first appearing at the top of the screen.

I don't really know why there is a delay in locating the tooltip in the right place on the page. But that will need more time to investigate. I'll create a GH issue about it.

**Pull request merged**